rfc/src/rfcs/0008-viewerless-coverage.md

```diff
@@ -0,0 +1,92 @@
+```
+<file>, <line>, <status>
+```
+where `<status>` is either `COVERED` or `UNCOVERED`.
+
+**Users are not expected to consume this output directly.**
+Instead, coverage data is to be consumed by the [Kani VS Code extension](https://github.com/model-checking/kani-vscode-extension) and displayed as in the following picture:
+
+![Coverage reported on the Kani VS Code Extension](../images/0008/coverage-kani-vscode.png)
+
+How to activate and display coverage information in the extension is out of scope for this RFC.
+We will need to change our documentation (including the tutorial) to use this option if we ever remove `--visualize` in favor of this one.
+
+## Detailed Design
+
+We will add a new unstable `--coverage` verification option to Kani.
+This will prevent `kani-driver` from reporting verification results as usual and instead output the coverage information described above[^coverage-assertions].
+It's likely that this will be done through a new output format that's not exposed to users.
+
+We will also add a new `--coverage-checks` option to `kani-compiler`, which will result in the injection of coverage checks before each Rust statement and terminator[^coverage-experiments].
+This option will be supplied by `kani-driver` when the `--coverage` option is selected.
+These coverage checks can be processed after regular verification postprocessing to obtain the coverage information, which can directly be passed on to the Kani VS Code Extension.
+
+Eventually, this could allow us to remove `cbmc-viewer` as a dependency.
+
+We anticipate missing locations being the main corner case for the coverage feature.
+However, coverage reported through `cbmc-viewer` also suffers from this issue.
+This will be mitigated by ignoring checks coming from unknown locations and propagating locations if possible whenever they're missing (as suggested in [this comment](https://github.com/model-checking/kani/issues/2048#issuecomment-1599680694)).
+
+## Rationale and alternatives
+
+The main advantage of this design is that it'll give Kani full control over its coverage-related options.
+This implies that not only it'll be able to select the mechanism to collect coverage information, but also have different options to output coverage data.
+All of that fully integrated in Kani and tested through `compiletest`.
+In contrast, this means that Kani will become more complex, so the development and maintainance burden will be higher.
+
+This feature is expected to have a substantial impact for Kani.
+Users frequently ask for coverage-related options but Kani doesn't have one.
+That's because, some months ago, we noticed incorrect coverage results were being reported with the `--visualize` option, prompting us to disable the coverage reports emitted by `cbmc-viewer`.
+
+## Open questions
+
+Open questions:
+ * Do we want a `--coverage` option or a `coverage` subcommand? Coverage can be seen as a different workflow but also a verification option.
```

Comment:
I think it depends whether verification will be performed at the same time.

Reply:
The RFC now proposes doing verification and coverage at the same time. My intuition says this will too much for harder problems, but we need to perform a proper evaluation (included in "feedback to gather before stabilization").